backend/supabase/migrations/20240814091741_notion-sync.sql

```diff
@@ -0,0 +1,83 @@
+drop index if exists "public"."knowledge_file_sha1_hash_idx";
+
+drop index if exists "public"."vectors_file_sha1_idx";
+
+create table "public"."knowledge_brain" (
+    "id" bigint generated by default as identity not null,
+    "created_at" timestamp with time zone not null default now(),
+    "knowledge_id" uuid,
+    "brain_id" uuid
+);
+
+
+alter table "public"."knowledge_brain" enable row level security;
+
+alter table "public"."vectors" drop column "file_sha1";
+
+alter table "public"."vectors" add column "knowledge_id" uuid;
+
+alter table "public"."vectors" alter column "id" set default gen_random_uuid();
+
+CREATE INDEX knowledge_brain_brain_id_idx ON public.knowledge_brain USING btree (brain_id);
+
+CREATE INDEX knowledge_brain_knowledge_id_idx ON public.knowledge_brain USING btree (knowledge_id);
+
+CREATE UNIQUE INDEX knowledge_brain_pkey ON public.knowledge_brain USING btree (id);
+
+CREATE INDEX vectors_knowledge_id_idx ON public.vectors USING btree (knowledge_id);
+
+alter table "public"."knowledge_brain" add constraint "knowledge_brain_pkey" PRIMARY KEY using index "knowledge_brain_pkey";
+
+alter table "public"."knowledge_brain" add constraint "public_knowledge_brain_brain_id_fkey" FOREIGN KEY (brain_id) REFERENCES brains(brain_id) ON UPDATE CASCADE ON DELETE CASCADE not valid;
+
+alter table "public"."knowledge_brain" validate constraint "public_knowledge_brain_brain_id_fkey";
+
+alter table "public"."knowledge_brain" add constraint "public_knowledge_brain_knowledge_id_fkey" FOREIGN KEY (knowledge_id) REFERENCES knowledge(id) ON UPDATE CASCADE ON DELETE CASCADE not valid;
+
+alter table "public"."knowledge_brain" validate constraint "public_knowledge_brain_knowledge_id_fkey";
+
+alter table "public"."vectors" add constraint "public_vectors_knowledge_id_fkey" FOREIGN KEY (knowledge_id) REFERENCES knowledge(id) ON UPDATE CASCADE ON DELETE CASCADE not valid;
+
+alter table "public"."vectors" validate constraint "public_vectors_knowledge_id_fkey";
```

Comment:
This query should wait for data insertion before running validation on knowledge_id column